### ISSUE TITLE
Lost focus when filling in dates

### ISSUE THREAD

**Issue by DOFfactory** [JabRef/jabref#4500]
JabRef 5.0-dev--snapshot--2018-11-18--master--64f320971
Windows 10 10.0 amd64 
Java 1.8.0_172

It is extremely frustrating when one tries to fill in a field `Date`, and I start with `2018-` and the focus drops to the `Author` field and one basically rewrites the already filled in `Author` field. The focus is also lost in the entry list panel, I have to search for the highlighted line - which is lost as well, and I need to use the search box for the title in the entry list. 

The same lost focus happens when one tries to fill in the keyword, which often fails to automatically generate itself. Instead, the keyword, if defined, should automatically fill in as the required fields are filled in prospectively - *without losing focus of the actually filled in field.*

**Comment by DOFfactory** [comment]
Ok, I believe I identified what is the issue here. The problem is the format one tries to insert the date. It is expecting the format `DD-MM-YYYY`, and once typed in this way, it "reformats" the date to the biblatex standard `YYYY-MM-DD`. This way the typing does not lose focus and does not jump to the Author field.

However, if one is aware of the biblatex date standard and starts to type it in that way, i.e. `YYYY-MM-DD`, after `YY` the focus is indeed lost and jumps to the Author field, rewriting it. 

I am not aware why is this happening, but it is a rather silly bug, to say the least. The expected behaviour would be not to lose focus, and accept both formats (`YYYY-MM-DD` and `DD-MM-YYYY`) and then reformat it, if needed for meeting the biblatex standard.

**Comment by Siedlerchr** [comment]
I can not reproduce this in the current master. I tried both 2018-12-21 and 21.12.2018 as example dates.
The date only gets reformatted if I leave the field with tabs

**Comment by DOFfactory** [comment]
Yes, I can confirm that this issue is not present anymore in the newest build. Thanks for looking into this.
